src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -169,14 +171,113 @@ private function createRelationsForItem($item, $formattedRelations)
+     * - `fallback_id` could be provided. In this case instead of deleting we set the connecting key to whatever developer gives us.
+     *
+     * @return void
+     */
+    private function attachManyRelation($item, $relation, $relationDetails, $relation_values)
+    {
+        $model_instance = $relation->getRelated();
+        $relation_foreign_key = $relation->getForeignKeyName();
+        $relation_local_key = $relation->getLocalKeyName();
+
+        if ($relation_values !== null) {
+            // we add the new values into the relation
+            $model_instance->whereIn($model_instance->getKeyName(), $relation_values)
+                ->update([$relation_foreign_key => $item->{$relation_local_key}]);
+
+            // we clear up any values that were removed from model relation.
+            // if developer provided a fallback id, we use it
+            // if column is nullable we set it to null if developer didn't specify `force_delete => true`
+            // if none of the above we delete the model from database
+            $removed_entries = $model_instance->whereNotIn($model_instance->getKeyName(), $relation_values)
+                                ->where($relation_foreign_key, $item->{$relation_local_key});
+
+            $this->handleManyRelationItemRemoval($model_instance, $removed_entries, $relationDetails, $relation_foreign_key);
+        } else {
+            // the developer cleared the selection
+            // we gonna clear all related values by setting up the value to the fallback id, to null or delete.
+            $removed_entries = $model_instance->where($relation_foreign_key, $item->{$relation_local_key});
+            $this->handleManyRelationItemRemoval($model_instance, $removed_entries, $relationDetails, $relation_foreign_key);
+        }
+    }
+
+    private function handleManyRelationItemRemoval($model_instance, $removed_entries, $relationDetails, $relation_foreign_key)
+    {
+        $relation_column_is_nullable = $model_instance->isColumnNullable($relation_foreign_key);
+        $force_delete = $relationDetails['force_delete'] ?? false;
+        $fallback_id = $relationDetails['fallback_id'] ?? false;
+
+        if ($fallback_id) {
+            return $removed_entries->update([$relation_foreign_key => $fallback_id]);
+        }
+
+        if ($force_delete) {
+            return $removed_entries->delete();
+        }
+
+        if (! $relation_column_is_nullable && $model_instance->dbColumnHasDefault($relation_foreign_key)) {
+            return $removed_entries->update([$relation_foreign_key => $model_instance->getDbColumnDefault($relation_foreign_key)]);
+        }
+
+        return $removed_entries->update([$relation_foreign_key => null]);
+    }
+
+    /**
+     * Handle HasMany/MorphMany relations when used as creatable entries in the crud.
+     * By using repeatable field, developer can allow the creation of such entries
+     * in the crud forms.
```

Comment:
I've tested this and... as it is right now... I think it poses a serious problem to data integrity. Every time the "monster" form is saved, the IDs on the related HasMany entries will be changed. No matter if there have been changes to the actual `HasMany` repeatable input or not.

![2021-12-21 13 59 03](https://user-images.githubusercontent.com/1032474/146926577-53641a2d-c6ad-4605-926c-9bd1e6947e18.gif)

This is not ok. From what I understand, this might be done in order to reorder them... Which is a tricky thing to achieve without an `order` column. But this current implementation is not ok. IDs should never be changed. And especially not from a related entity.

I know we said this is _probably_ going to be used for subitems that are completely dependant on the main item, but that doesn't mean OTHER items could not depend on this subitem.

Here's an example:
- Invoice HasMany InvoiceItems
- License BelongsTo InvoiceItem (in order to get the purchase price, purchaser, etc) 

So the `licenses` table has a column `invoice_item_id`. If that ID gets changed on the `invoice_items` table, when you edit your `invoice`... You get an orphaned License.

Any reason to do this @pxpm ?

Reply:
What's particularly odd is that this DOES happen for `HasMany` in the 4.2 demo, but does NOT happen for the `MorphMany` in the demo. Even though I believed they both ran the same code to do this. Notice I will reorder both:
- MorphMany reordering will not be saved
- HasMany reordering will be saved (because IDs get changed)

![2021-12-21 14 16 36](https://user-images.githubusercontent.com/1032474/146928690-ec2520f2-e3e8-48a3-b747-695a04fe6e70.gif)

Reply:
Again, I think it is WAAAAY more important here for the IDs not to get changed. We can disable the reorder functionality in this use case, for what I care. But the IDs should stay the same.

Reply:
Totally agree with you. 

Although it added the "ability" to order, was not an intentional behavior and a nasty bug, I 100% agree with you that keys should not change. 

I think I fixed that behaviour, and added tests to account for the key too, I was only testing the value of other fields. 

Give it a go please and let me know if it's fixed. Ofcourse, no order for now.

Reply:
Yup, works fine now! I've pushed a change to `repeatable` on this branch, so we can turn off the reordering. And turned off the reordering inside `repeatable_relation`. So the field is no longer confusing now - it does what it seems to do. And it's no longer dangerous now - it does not increment the IDs 🎉🎉🎉

Good work team 😅

Reply:
It's sad that we don't have the reorder functionality here too... 😔 I've added my idea on how to implement this to https://github.com/Laravel-Backpack/CRUD/issues/4012